tests/Unit/UseCases/ApplyForMembership/MembershipApplicationValidatorTest.php

```diff
@@ -50,12 +48,57 @@ public function testGivenInvalidAmount_validationFails( string $amount, string $
 
 	public function invalidAmountProvider() {
 		return [
-			'too low' => [ '1.00', Result::VIOLATION_TOO_LOW ],
-			'just too low' => [ '24.99', Result::VIOLATION_TOO_LOW ],
-			'max too low' => [ '0', Result::VIOLATION_TOO_LOW ],
-			'invalid: negative' => [ '-1.00', Result::VIOLATION_NOT_MONEY ],
-			'invalid' => [ 'y u no btc', Result::VIOLATION_NOT_MONEY ],
+			'invalid: negative' => [ '-1.00', 3, Result::VIOLATION_NOT_MONEY ],
+			'invalid' => [ 'y u no btc', 3, Result::VIOLATION_NOT_MONEY ],
+
+			'too low single payment' => [ '1.00', 12, Result::VIOLATION_TOO_LOW ],
+			'just too low single payment' => [ '23.99', 12, Result::VIOLATION_TOO_LOW ],
+			'max too low single payment' => [ '0', 12, Result::VIOLATION_TOO_LOW ],
+
+			'too low 12 times' => [ '1.99', 1, Result::VIOLATION_TOO_LOW ],
+			'too low 4 times' => [ '5.99', 3, Result::VIOLATION_TOO_LOW ],
+		];
+	}
+
+	private function newValidRequestWithPaymentAmount( string $amount, int $intervalInMonths ) {
+		$request = $this->newValidRequest();
+		$request->setPaymentAmountInEuros( $amount );
+		$request->setPaymentIntervalInMonths( $intervalInMonths );
+		return $request;
+	}
+
+	/**
+	 * @dataProvider validAmountProvider
+	 */
+	public function testGivenValidAmount_validationSucceeds( string $amount, int $intervalInMonths ) {
+		$request = $this->newValidRequestWithPaymentAmount( $amount, $intervalInMonths );
+
+		$this->assertTrue( $this->newValidator()->validate( $request )->isSuccessful() );
+	}
+
+	public function validAmountProvider() {
+		return [
+			'single payment' => [ '50.00', 12 ],
+			'just enough single payment' => [ '24.00', 12 ],
+			'high single payment' => [ '31333.37', 12 ],
+
```

Comment:
These are amounts with decimal points instead of comma. Should't we be consistent with the donation page and expect comma-formatted amounts (depending on locale)?

And maybe you can add an integer amount which should also be valid.